### ISSUE TITLE
[clang-tidy] False positive bugprone-non-zero-enum-to-bool-conversion with custom operator and negation

### ISSUE THREAD

**Issue by chrchr-github** [llvm/llvm-project#81515]
A slight variation of [#65315](https://github.com/llvm/llvm-project/issues/65315)
~~~c++
enum E { E1 = 1, E2 = 2 };
E operator&(E a, E b) { return static_cast<E>((int)a & (int)b); }
void f(E e) {
    if (!(e & E1)) {}
}
~~~
~~~
[<source>:4:10: warning: conversion of 'E' into 'bool' will always return 'true', enum doesn't have a zero-value enumerator [bugprone-non-zero-enum-to-bool-conversion]
    4 |     if (!(e & E1)) {}
      |          ^
[<source>:1:6: note: enum is defined here](javascript:;)
    1 | enum E { E1 = 1, E2 = 2 };
      |      ^
~~~
https://godbolt.org/z/r1P3zx38a

**Comment by PiotrZSL** [comment]
Problem is caused by extra ().
Fix:
```
--- a/clang-tools-extra/clang-tidy/bugprone/NonZeroEnumToBoolConversionCheck.cpp
+++ b/clang-tools-extra/clang-tidy/bugprone/NonZeroEnumToBoolConversionCheck.cpp
@@ -64,7 +64,7 @@ void NonZeroEnumToBoolConversionCheck::registerMatchers(MatchFinder *Finder) {
                                          EnumIgnoreList)))
                                 .bind("enum"))))),
                         unless(declRefExpr(to(enumConstantDecl()))),
-                        unless(ignoringImplicit(ExcludedOperators)))),
+                        unless(ignoringParenImpCasts(ExcludedOperators)))),
                unless(hasAncestor(staticAssertDecl())))
           .bind("cast"),
       this);
```
